### ISSUE TITLE
Package dx: UNEXPECTED TOP-LEVEL ERROR

### ISSUE THREAD

**Issue by aeuxer** [termux/termux-packages#1801]
Cannot generate dex file via dx. Running the following input:

rm -rf obj

mkdir -p obj

aapt package -v -f \
-M AndroidManifest.xml \
-J $(find . -type d -name obj) \
-S $(find . -type d -name res) \
-m

ecj -d ./obj -sourcepath . $(find . -type f \
-name "*.java")

dx --dex --verbose \
--output=./obj/classes.dex ./obj

Can generate R.java. Can compile Java source to class files. Get the following output when running dx step:

UNEXPECTED TOP-LEVEL ERROR:
java.lang.NoSuchFieldError: No instance field callSiteIds of type Lcom/android/dex/TableOfContents$Section; in class Lcom/android/dex/TableOfContents; or its superclasses (declaration of 'com.android.dex.TableOfContents' appears in /system/framework/core-libart.jar)                                                at com.android.dx.merge.IndexMap.<init>(IndexMap.java:84)
        at com.android.dx.merge.DexMerger.<init>(DexMerger.java:107)
        at com.android.dx.merge.DexMerger.<init>(DexMerger.java:93)                                           at com.android.dx.command.dexer.Main.mergeLibraryDexBuffers(Main.java:506)                            at com.android.dx.command.dexer.Main.runMonoDex(Main.java:336)
        at com.android.dx.command.dexer.Main.runDx(Main.java:291)

**Comment by sdrausty** [comment]
Ditto:
```
UNEXPECTED TOP-LEVEL ERROR:
java.lang.NoSuchFieldError: No instance field callSiteIds of type Lcom/android/dex/TableOfContents$Section; in class Lcom/android/dex/TableOfContents; or its superclasses (declaration of 'com.android.dex.TableOfContents' appears in /system/framework/core-libart.jar)
        at com.android.dx.merge.IndexMap.<init>(IndexMap.java:84)
        at com.android.dx.merge.DexMerger.<init>(DexMerger.java:107)
        at com.android.dx.merge.DexMerger.<init>(DexMerger.java:93)
        at com.android.dx.command.dexer.Main.mergeLibraryDexBuffers(Main.java:506)
        at com.android.dx.command.dexer.Main.runMonoDex(Main.java:336)
        at com.android.dx.command.dexer.Main.runDx(Main.java:291)
        at com.android.dx.command.dexer.Main.main(Main.java:247)
        at com.android.dx.command.Main.main(Main.java:94)
$ uname -mo
aarch64 Android
$ getprop ro.product.cpu.abi
arm64-v8a
```

**Comment by sdrausty** [comment]
```
UNEXPECTED TOP-LEVEL ERROR:
java.lang.NoSuchFieldError: No instance field callSiteIds of type Lcom/android/dex/TableOfContents$Section; in class Lcom/android/dex/TableOfContents; or its superclasses (declaration of 'com.android.dex.TableOfContents' appears in /system/framework/core-libart.jar)
        at com.android.dx.merge.IndexMap.<init>(IndexMap.java:84)
        at com.android.dx.merge.DexMerger.<init>(DexMerger.java:107)
        at com.android.dx.merge.DexMerger.<init>(DexMerger.java:93)
        at com.android.dx.command.dexer.Main.mergeLibraryDexBuffers(Main.java:506)
        at com.android.dx.command.dexer.Main.runMonoDex(Main.java:336)
        at com.android.dx.command.dexer.Main.runDx(Main.java:291)
        at com.android.dx.command.dexer.Main.main(Main.java:247)
        at com.android.dx.command.Main.main(Main.java:94)
$ uname -mo
armv7l Android
$ getprop ro.product.cpu.abi
armeabi-v7a
```
Link to older version of  `dx` https://github.com/sdrausty/repoAll/blob/master/dists/termux/extras/binary-all/dx_26.0.1_all.deb at https://sdrausty.github.io/repoAll/

**Comment by fornwall** [comment]
Thanks - this should now be fixed in version `27.0.1-1` of the dx package - update to it by running `pkg up`!

**Comment by sdrausty** [comment]
Tested and charmingly work on both platforms once again.  Thank you for your time and the fix @fornwall :)
